main.py

```diff
@@ -48,6 +48,41 @@ def parallel_feature_extraction(args):
     extractor.progress.close()
 
 
+def cpu_feature_extraction(args):
```

Comment:
This is great!

Hmm, could we make this a single function? I would like to avoid code duplication.

For instance, something like
```python
if args.cpu:
   do this
else:
   do that
```

The function name will be misleading but that's ok for now.